Modules/_json.c

```diff
@@ -1491,17 +1492,73 @@ encoder_listencode_obj(PyEncoderObject *s, _PyUnicodeWriter *writer,
     }
 }
 
+static int
+encoder_encode_key_value(PyEncoderObject *s, _PyUnicodeWriter *writer, bool *first,
+                         PyObject *key, PyObject *value, Py_ssize_t indent_level)
+{
+    PyObject *keystr = NULL;
+    PyObject *encoded;
+
+    if (PyUnicode_Check(key)) {
+        Py_INCREF(key);
+        keystr = key;
+    }
+    else if (PyFloat_Check(key)) {
+        keystr = encoder_encode_float(s, key);
+    }
+    else if (key == Py_True || key == Py_False || key == Py_None) {
+                    /* This must come before the PyLong_Check because
+                       True and False are also 1 and 0.*/
+        keystr = _encoded_const(key);
+    }
+    else if (PyLong_Check(key)) {
+        keystr = PyLong_Type.tp_repr(key);
+    }
+    else if (s->skipkeys) {
+        return 0;
+    }
+    else {
+        PyErr_Format(PyExc_TypeError,
+                     "keys must be str, int, float, bool or None, "
+                     "not %.100s", Py_TYPE(key)->tp_name);
+        return -1;
+    }
+
+    if (keystr == NULL)
+        return -1;
+
+    if (*first) {
+        *first = false;
+    } else {
+        if (_PyUnicodeWriter_WriteStr(writer, s->item_separator)) {
+            Py_DECREF(keystr);
+            return -1;
+        }
+    }
+
+    encoded = encoder_encode_string(s, keystr);
+    Py_DECREF(keystr);
+    if (encoded == NULL)
+        return -1;
+
+    if (_steal_accumulate(writer, encoded) < 0)
+        return -1;
+    if (_PyUnicodeWriter_WriteStr(writer, s->key_separator))
+        return -1;
+    if (encoder_listencode_obj(s, writer, value, indent_level))
```

Comment:
```suggestion
    if (encoder_listencode_obj(s, writer, value, indent_level) < 0)
```

For consistency, nit